edi_oca/models/edi_exchange_record.py

```diff
@@ -209,7 +211,13 @@ def record(self):
             return None
         if not self.model and self.parent_id:
             return self.parent_id.record
-        return self.env[self.model].browse(self.res_id).exists()
+        rec = self.env[self.model].browse(self.res_id).exists()
+        if rec and "company_id" in rec:
+            rec_company = rec.sudo().company_id
+            if rec_company and rec_company not in self.env.companies:
+                # Return empty recordset if company does not match
+                rec = self.env[self.model].browse()
+        return rec
```

Comment:
Just to understand: why not use a `search()` and let Odoo filter out the record if it doesn't match the proper multicompany security rules?
```suggestion
        return self.env[self.model].sudo(False).search([("id", "=", int(self.res_id))])
```

Reply:
@SilvioC2C do you think that there is something seriously breaking with current code? to me current implementation says for itself what we try to prevent here.

Reply:
What I mean is: multicompany rules can be defined in different ways. For example, we may have different rules that test the record's company against the current user's main company (`self.env.user.company_id`), user's allowed companies (`self.env.user.company_ids`), Env's company (`self.env.company`), session's currently enabled companies (`self.env.companies`), etc...
Using `search()` will make Odoo test all these cases, while this current check only tests 1 case.

Reply:
@SilvioC2C  updated.